OpenRA.Mods.Common/ServerTraits/LobbyCommands.cs

```diff
@@ -83,682 +83,736 @@ public bool InterpretCommand(S server, Connection conn, Session.Client client, s
 
 			var dict = new Dictionary<string, Func<string, bool>>
 			{
-				{ "state",
-					s =>
-					{
-						var state = Session.ClientState.Invalid;
-						if (!Enum<Session.ClientState>.TryParse(s, false, out state))
-						{
-							server.SendOrderTo(conn, "Message", "Malformed state command");
-							return true;
-						}
+				{ "state", State(server, conn, client) },
+				{ "startgame", StartGame(server, conn, client) },
+				{ "slot", Slot(server, conn, client) },
+				{ "allow_spectators", AllowSpectators(server, conn) },
+				{ "spectate", Specate(server, client) },
+				{ "slot_close", SlotClose(server, conn, client) },
+				{ "slot_open", SlotOpen(server, conn, client) },
+				{ "slot_bot", SlotBot(server, conn, client) },
+				{ "map", Map(server, conn, client) },
+				{ "option", Option(server, conn, client) },
+				{ "assignteams", AssignTeams(server, conn, client) },
+				{ "kick", Kick(server, conn, client) },
+				{ "name", Name(server, conn, client) },
+				{ "faction", Faction(server, conn, client) },
+				{ "team", Team(server, client) },
+				{ "spawn", Spawn(server, conn, client) },
+				{ "color", Color(server, conn, client) },
+				{ "sync_lobby", SyncLobby(server, conn, client) }
+			};
 
-						client.State = state;
+			var cmdName = cmd.Split(' ').First();
+			var cmdValue = cmd.Split(' ').Skip(1).JoinWith(" ");
 
-						Log.Write("server", "Player @{0} is {1}",
-							conn.Socket.RemoteEndPoint, client.State);
+			Func<string, bool> a;
+			if (!dict.TryGetValue(cmdName, out a))
+				return false;
 
-						server.SyncLobbyClients();
+			return a(cmdValue);
+		}
 
-						CheckAutoStart(server);
+		static Func<string, bool> SyncLobby(S server, Connection conn, Session.Client client)
```

Comment:
Can you please reorder these to match the original switch (i.e. the order from the dict definition above).  This will help git identify the real changes in this file and preserve the git history, instead of overriding it with a bunch of misattributed additions/deletions.